### PR TITLE
[wip] Add Ironic MaaS checks and alarms

### PR DESCRIPTION
**Pull request by cfarquhar**
- Adds a feature flag to include ironic-api and ironic-conductor in an AIO deployment.
- Adds a local and remote check for the Ironic API

Connects https://github.com/rcbops/u-suk-dev/issues/91

### CODE REVIEW

**Comment by cfarquhar** [comment]
Waiting on https://github.com/rcbops/rpc-openstack/pull/1354 or an alternative approach to creating MaaS checks conditionally.

File: scripts/deploy.sh [hunk 1]
```diff
@@ -78,6 +79,12 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
     # set network speed for vms
     echo "net_max_speed: 1000" >>$RPCD_VARS
 
+    # Copy ironic AIO config if ironic is enabled
+    if [[ "$DEPLOY_IRONIC" == "yes" ]]; then
+      cp -a -a ${OA_DIR}/etc/openstack_deploy/conf.d/ironic.yml.aio /etc/openstack_deploy/conf.d/ironic.yml
+      cp -a -a ${OA_DIR}/etc/openstack_deploy/env.d/ironic.yml /etc/openstack_deploy/env.d/
```
Comment (mrda) [inline scripts/deploy.sh]: Should we also be setting os_nova/defaults/main.yml:nova_virt_type to ironic as well?

File: scripts/deploy.sh [hunk 2]
```diff
@@ -78,6 +79,12 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
     # set network speed for vms
     echo "net_max_speed: 1000" >>$RPCD_VARS
 
+    # Copy ironic AIO config if ironic is enabled
+    if [[ "$DEPLOY_IRONIC" == "yes" ]]; then
+      cp -a -a ${OA_DIR}/etc/openstack_deploy/conf.d/ironic.yml.aio /etc/openstack_deploy/conf.d/ironic.yml
```
Comment (mrda) [inline scripts/deploy.sh]: Why do both of these cp commands have "-a" twice?

Comment (cfarquhar) [inline scripts/deploy.sh]: Thanks for the review @mrda.  Somehow your review was against an "outdated diff".  The extra flag was removed in the last PR.

File: scripts/deploy.sh (hunk 1 above)
Comment (cfarquhar) [inline scripts/deploy.sh]: Yes, good catch.  I'll fix this shortly.

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/staging.horse/build-af701d743b/docs/private-cloud/release-notes/). :bow:

**Comment by Frank-ZhangXin** [comment]
recheck_ceph

**Comment by alextricity25** [comment]
Thanks @cfarquhar  for this work! It's highly appreciated. Ironic is still currently being discussed as an item in our roadmap and what release it will be a part of, but I don't want to lose track of this PR.
I would feel more comfortable if these changes were made once Ironic is a little closer to being integrated with RPCO. I'll keep this PR open for visibility, and connect it to https://github.com/rcbops/u-suk-dev/issues/91

**Comment by cfarquhar** [comment]
Hey @alextricity25 .  I still need to get these (or some equivalent) changes in.  I'm just standing by for the repo to be ready to accept them.  Can we leave this open, or is there a reason to open another PR?

**Comment by alextricity25** [comment]
Thanks for the heads up @cfarquhar. I"ll reopen the PR for reviews.

**Comment by odyssey4me** [comment]
I think the rpc-maas repo is now ready for patch submissions, so this PR should be closed and a PR opened there.

**Comment by cfarquhar** [comment]
Agreed.  Getting a PR to the new rpc-maas repo is on my to-do list for today.  Thanks!

**Pull request closed without merging**